From 058d5e8cd90d117535b5d314fd9e15c70028ff20 Mon Sep 17 00:00:00 2001
From: Eric Le Bihan <eric.le.bihan.dev@free.fr>
Date: Mon, 21 Apr 2014 19:31:23 +0200
Subject: [PATCH] Add missing GLchar definitions.

The GLES and GLES2 header files provided by sunxi-mali do not define the
type "GLchar" and use "char" instead in the prototype of some functions.

This issue has been reported upstream some time ago, but the pull
request has not been merged yet.  (See
https://github.com/linux-sunxi/sunxi-mali/pull/8).

Signed-off-by: Eric Le Bihan <eric.le.bihan.dev@free.fr>
---
 include/GLES/gl.h   | 1 +
 include/GLES2/gl2.h | 1 +
 2 files changed, 2 insertions(+)

diff --git a/include/GLES/gl.h b/include/GLES/gl.h
index 858f394..a6bb591 100644
--- a/include/GLES/gl.h
+++ b/include/GLES/gl.h
@@ -29,6 +29,7 @@ typedef float  GLfloat;
 typedef float  GLclampf;
 typedef signed int  GLfixed;
 typedef signed int  GLclampx;
+typedef char GLchar;
 
 typedef int * GLintptr;
 typedef int *  GLsizeiptr;
diff --git a/include/GLES2/gl2.h b/include/GLES2/gl2.h
index 59e376c..90d96bb 100644
--- a/include/GLES2/gl2.h
+++ b/include/GLES2/gl2.h
@@ -32,6 +32,7 @@ typedef unsigned int     GLuint;
 typedef khronos_float_t  GLfloat;
 typedef khronos_float_t  GLclampf;
 typedef khronos_int32_t  GLfixed;
+typedef char             GLchar;
 
 /* GL types for handling large vertex buffer objects */
 typedef khronos_intptr_t GLintptr;
-- 
1.9.0

